src/RequestConditionalPlugin.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Http\Message\RequestMatcher;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Apply a delegated plugin based on a request match.
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+final class RequestConditionalPlugin implements Plugin
```

Comment:
i am not too happy with the name. the name does not really tell how the 3 terms go together. is this a plugin to request a conditional? or to make the request conditional?

but unfortunately i have no better name. ConditionalRequestPlugin certainly is worse.